src/documents/templates/ParentAndChildCodeLenses.ts

```diff
@@ -0,0 +1,123 @@
+// ---------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.md in the project root for license information.
+// ---------------------------------------------------------------------------------------------
+
+import { Location } from "vscode";
+import { configKeys } from "../../constants";
+import { ext } from "../../extensionVariables";
+import { sortArrayByProperty } from "../../util/sortArrayByProperty";
+import { IPeekResourcesArgs } from "../../vscodeIntegration/commandArguments";
+import { getVSCodeRangeFromSpan } from "../../vscodeIntegration/vscodePosition";
+import { ResolvableCodeLens } from "../DeploymentDocument";
+import { IJsonResourceInfo } from "./getResourcesInfo";
+import { TemplateScope } from "./scopes/TemplateScope";
+
+export function getParentAndChildCodeLenses(scope: TemplateScope, infos: IJsonResourceInfo[]): ResolvableCodeLens[] {
+    if (!ext.configuration.get<boolean>(configKeys.codeLensForResourceParentsAndChildren)) {
+        return [];
+    }
+
+    const lenses: ResolvableCodeLens[] = [];
+
+    for (const resource of infos) {
+        lenses.push(new ParentCodeLens(scope, resource));
+        lenses.push(new ChildrenCodeLens(scope, resource));
+    }
+
+    return lenses;
+}
+
+export abstract class ParentOrChildCodeLens extends ResolvableCodeLens {
+    public constructor(
+        scope: TemplateScope,
+        protected readonly sourceResource: IJsonResourceInfo,
+    ) {
+        super(scope, sourceResource.resourceObject.span);
+    }
+
+    protected resolveCore(title: string, targets: IJsonResourceInfo[], kind: 'parent' | 'children'): boolean {
+        if (targets.length > 0) {
+            const targetLocations = targets.map(resInfo =>
+                new Location(
+                    this.scope.document.documentUri,
+                    getVSCodeRangeFromSpan(this.scope.document, resInfo.resourceObject.span))
+            );
+            this.command = {
+                title: title,
+                command: 'azurerm-vscode-tools.codeLens.peekResources',
+                arguments:
+                    [
+                        <IPeekResourcesArgs>{
+                            source: { uri: this.scope.document.documentUri, range: this.range },
+                            targets: targetLocations,
+                            telemetryProperties: {
+                                kind,
+                                targetCount: String(targets.length)
+                            }
+                        }
+                    ]
+            };
+        } else {
+            this.command = {
+                title: title,
```

Comment:
Just curious. How is this rendered and used?

Reply:
See the images in the description, e.g.:
![image](https://user-images.githubusercontent.com/6913354/93540473-9ef2fa80-f908-11ea-917a-dd76781f982d.png)

The command title is the "No parent" or "2 children: xxx" text.  If you click on a command with children or parent, it peeks them in the editor.
![image](https://user-images.githubusercontent.com/6913354/94218884-54d6bf80-fe9a-11ea-96f7-81700852f267.png)